### PR TITLE
Create Messaging Topology

### LINKED ISSUE #6004: Create Messaging Topology

### PR DESCRIPTION
**Pull request by MilanBariya**
### Describe your changes :

I worked on Create Messaging Topology
Fix #6004 
#
### Type of change :

- [x] Bug fix
- [x] Improvement

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/messaging/kafka.py [hunk 1]
```diff
@@ -100,90 +79,77 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
...
-                self.status.failure(topic_name, repr(err))
+            yield KafkaTopicDetails(topic_name, topic_metadata)
+
+    def get_topic_name(self, topic_details: KafkaTopicDetails) -> str:
+        """
+        Get Topic Name
+        """
+        return topic_details.topic_name
+
+    def yield_topic(self, topic_details: KafkaTopicDetails) -> Any:
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/messaging/kafka.py:93]: we are yielding a Topic here so typing can be changed to Topic

File: ingestion/src/metadata/ingestion/source/messaging/kafka.py [hunk 2]
```diff
@@ -100,90 +79,77 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
...
+            ),
+        )
+        topic_configResource = self.admin_client.describe_configs(
+            [
+                ConfigResource(
+                    confluent_kafka.admin.RESOURCE_TOPIC, topic_details.topic_name
+                )
+            ]
+        )
+        for j in concurrent.futures.as_completed(iter(topic_configResource.values())):
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/messaging/kafka.py:114]: Are we creating a ThreadPool somewhere?

File: ingestion/src/metadata/ingestion/source/messaging/kafka.py [hunk 3]
```diff
@@ -34,57 +29,41 @@
...
-    def topic_scanned(self, topic: str) -> None:
-        self.topics_scanned.append(topic)
+    topic_name: str
+    topic_metadata: Any
 
-    def dropped(self, topic: str) -> None:
-        self.filtered.append(topic)
+    def __init__(self, topic_name: str, topic_metadata: Any) -> None:
+        self.topic_name = topic_name
+        self.topic_metadata = topic_metadata
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/messaging/kafka.py:56]: Should that be defined as a jsonschema or at least using pydantic for consistency?

File: ingestion/src/metadata/ingestion/source/messaging/messaging_service.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                type_=Topic,
+                context="Topic",
+                processor="yield_topic",
+                consumer=["messaging_service"],
+            )
+        ],
+    )
+
+
+@dataclass
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/messaging/messaging_service.py:87]: Do you know why we use a dataclass here?

Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/messaging/messaging_service.py:87]: `class SourceStatus(Status):` is using dataclass so I am using the same approach.

File: ingestion/src/metadata/ingestion/source/messaging/kafka.py (hunk 3 above)
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/messaging/kafka.py:56]: @TeddyCr Using pydantic..will update in the next commit.

File: ingestion/src/metadata/ingestion/source/messaging/messaging_service.py (hunk 4 above)
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/messaging/messaging_service.py:87]: We have a task to overhaul the source status. We'll update to pydantic there https://github.com/open-metadata/OpenMetadata/issues/4694

File: catalog-rest-service/src/main/resources/json/schema/api/data/createTopic.json [hunk 5]
```diff
@@ -44,6 +44,11 @@
         "$ref": "../../entity/data/topic.json#/definitions/cleanupPolicy"
       }
     },
+    "sampleData": {
```
[the remark this answers is not in the document]
Comment (MilanBariya) [reply catalog-rest-service/src/main/resources/json/schema/api/data/createTopic.json:47]: @ulixius9 Yes it's needs to be handles from backend side as well

File: ingestion/src/metadata/ingestion/source/messaging/kafka.py (hunk 1 above)
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/messaging/kafka.py:93]: @ulixius9 sure i will do that.

File: ingestion/src/metadata/ingestion/source/messaging/kafka.py [hunk 6]
```diff
@@ -100,90 +79,77 @@ def create(cls, config_dict, metadata_config: OpenMetadataConnection):
...
+                topic.cleanupPolicies = cleanup_policies.split(",")
+
+            topic_config = {}
+            for key, conf_response in config_response.items():
+                topic_config[key] = conf_response.value
+            topic.topicConfig = topic_config
+
+        if topic_schema is not None:
+            topic.schemaText = topic_schema.schema_str
+            if topic_schema.schema_type == "AVRO":
```
[the remark this answers is not in the document]
Comment (MilanBariya) [reply ingestion/src/metadata/ingestion/source/messaging/kafka.py:140]: @ulixius9 sure will update.

**Review by pmbrull — APPROVED** [review]

**Pull request merged**